storage/config.go

```diff
@@ -52,8 +52,9 @@ type Config struct {
 	// inputs is accelerated. The size of the cache is large so understand how to configure the cache.
 	// TODO provide some default config choices
 	// If this section is skipped, Caching is disabled
-	Cache  CachingConfig `json:"cache"`
-	Limits LimitsConfig  `json:"limits" pflag:",Sets limits for stores."`
+	Cache                    CachingConfig       `json:"cache"`
+	Limits                   LimitsConfig        `json:"limits" pflag:",Sets limits for stores."`
+	DefaultHTTPClientHeaders map[string][]string `json:"defaultHttpClientHeaders" pflag:"-,Sets http headers to set on the default http client."`
```

Comment:
do you want to create a separate section called client-defaults?
headers: ....
etc